qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/AvailabilityTest.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.broker.it.clustering;
+
+import static io.zeebe.test.util.TestUtil.waitUntil;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.zeebe.broker.it.util.GrpcClientRule;
+import io.zeebe.client.api.response.ActivatedJob;
+import io.zeebe.client.api.response.BrokerInfo;
+import io.zeebe.model.bpmn.Bpmn;
+import io.zeebe.model.bpmn.BpmnModelInstance;
+import io.zeebe.protocol.record.Record;
+import io.zeebe.protocol.record.intent.JobIntent;
+import io.zeebe.protocol.record.intent.WorkflowInstanceCreationIntent;
+import io.zeebe.test.util.record.RecordingExporter;
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.Timeout;
+
+public class AvailabilityTest {
+
+  private static final String JOBTYPE = "availability-test";
+  private static final BpmnModelInstance WORKFLOW =
+      Bpmn.createExecutableProcess("process")
+          .startEvent()
+          .serviceTask(
+              "task",
+              t -> {
+                t.zeebeJobType(JOBTYPE);
+              })
+          .endEvent()
+          .done();
+  private final int partitionCount = 3;
+  private final Timeout testTimeout = Timeout.seconds(120);
+  private final ClusteringRule clusteringRule = new ClusteringRule(partitionCount, 1, 3);
+  private final GrpcClientRule clientRule = new GrpcClientRule(clusteringRule);
+
+  @Rule
```

Comment:
If we don't need a different broker every time, we can use a ClassRule, no?

Reply:
I was under the impression it can make test more complicated, but here I didn't think it would - if you think it would make it harder to read, then it's fine. My very weak argument here would be if we can use the ClassRule, better to do it at the start than later as it's more of a pain to refactor to use than to start with it immediately.